src/main/java/org/folio/search/service/converter/SearchFieldProcessor.java

```diff
@@ -0,0 +1,57 @@
+package org.folio.search.service.converter;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static org.folio.search.utils.SearchUtils.getMultilangValue;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.collections.MapUtils;
+import org.folio.search.model.metadata.SearchFieldDescriptor;
+import org.folio.search.service.setter.FieldProcessor;
+import org.folio.search.utils.JsonConverter;
+import org.springframework.stereotype.Component;
+
+@Log4j2
+@Component
+@RequiredArgsConstructor
+public class SearchFieldProcessor {
```

Comment:
```suggestion
public class SearchFieldProcessors {
```

Maybe add `s` to the class name?

Reply:
It's a really rare case to class to have a name in plural form, but I think, you have a point and I will rename it to the `SearchFieldsProcessor` to mark that is not a single field processor